Feature/search for archived projects

artem-sidorenko opened the pull request:
For admins it might be useful to see the archived projects in the project view of admin interface. Currently the archived projects **are not displayed** at all. The idea is to add the option to show archived projects and to label them in the result list.

![admin-projects](https://cloud.githubusercontent.com/assets/1645670/9179196/3b5eee42-3f9b-11e5-9661-27c0d0c3df4f.png)

For group owners it might be useful too. Currently the archived projects **are displayed** without any label or "archived hint". The idea is to have a label like in the admin interface in the result list.

![group-projects](https://cloud.githubusercontent.com/assets/1645670/9179202/41d1def6-3f9b-11e5-8928-e59a3152189a.png)

There is another small change, to have the same styled visibility icons in the admin interface and group project view.

See [this feature request](http://feedback.gitlab.com/forums/176466-general/suggestions/6982953-view-list-of-archived-projects)

artem-sidorenko committed:
Allow listing of archived projects

Resolves #9285

artem-sidorenko committed:
Show label on archived projects

artem-sidorenko committed:
Show colored icons like in the admin listing

stanhu commented:
Looks good to me. You may have to rebase master to resolve the current conflicts.

artem-sidorenko commented:
@stanhu rebased

stanhu commented:
@artem-sidorenko Thanks. @DouweM, any comments on the design here?

DouweM commented on app/controllers/admin/projects_controller.rb:
```diff
@@ -9,6 +9,7 @@ def index
     @projects = @projects.where("visibility_level IN (?)", params[:visibility_levels]) if params[:visibility_levels].present?
     @projects = @projects.with_push if params[:with_push].present?
     @projects = @projects.abandoned if params[:abandoned].present?
+    @projects = @projects.without_archived if !params[:with_archived].present?
```
Please use `unless` instead of `if !`

DouweM commented on app/models/project.rb:
```diff
@@ -210,12 +210,16 @@ def with_push
 
+    def without_archived
+      where('projects.archived = ?',false)
+    end
```
Can you use a `scope` for this instead?

DouweM commented on app/models/project.rb:
```diff
@@ -210,12 +210,16 @@ def with_push
 
     def search(query)
-      joins(:namespace).where('projects.archived = ?', false).
+      joins(:namespace).
```
In other places where we use `search`, we may still want the `.where('projects.archived = ?', false)`, like when regular users search. Could you check?

artem-sidorenko replied:
I found it in the `app/controllers/explore/projects_controller.rb` and added the non_archived scope

DouweM commented:
Looks all right design-wise, but the code leaves a bit to be desired :)

artem-sidorenko commented:
@DouweM thx for feedback:) updated in 51a7ad5

Question here about `app/controllers/explore/projects_controller.rb`:
 - I've excluded the archived projects in the "All projects" list
 - In the Trending or Starred lists they are displayed. Should this be this way or should the archived projects be hidden here?

artem-sidorenko committed:
Spinach tests

DouweM commented:
@artem-sidorenko They should be hidden under Trending as well. Starred is user-initiated, so that can include archived ones.

artem-sidorenko commented:
@DouweM can you please have a look again?

Razer6 commented on CHANGELOG:
```diff
@@ -3,6 +3,7 @@ Please view this file on the master branch, on stable branches it's out of date.
 v 7.14.0 (unreleased)
   - Disable turbolinks when linking to Bitbucket import status (Stan Hu)
   - Fix broken code import and display error messages if something went wrong with creating project (Stan Hu)
+  - Allow displaying of archived projects in the admin interface (Artem Sidorenko)
```
Because there is already design freeze for 7.14, this should go to 8.0

artem-sidorenko replied:
@Razer6 in the Changelog in master is no `v 8.0`. Should I create it and move it?

Razer6 replied:
I updated the changelog. If you rebase you can move it to 8.0

artem-sidorenko commented:
@Razer6 thx, done!
@DouweM I rebased on the latests master and squashed this MR to the several logical independent commits (probably it makes a review easier, I don't know). I hope its ok. Give me a ping if I should squash this MR to one single commit

DouweM commented:
@artem-sidorenko Looks good to me, merged :)

tsigo commented:
@artem-sidorenko Congrats on your first merge! Thanks for helping out.
